### PR TITLE
r/s3_bucket_website_configuration: new resource

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #4418
Relates #20433

Output from acceptance testing:

```
--- PASS: TestAccS3BucketWebsiteConfiguration_disappears (29.72s)
--- PASS: TestAccS3BucketWebsiteConfiguration_basic (33.03s)
--- PASS: TestAccS3BucketWebsiteConfiguration_RoutingRules_RedirectOnly (33.37s)
--- PASS: TestAccS3BucketWebsiteConfiguration_Redirect (33.38s)
--- PASS: TestAccS3BucketWebsiteConfiguration_update (51.82s)
--- PASS: TestAccS3BucketWebsiteConfiguration_RoutingRules_MultipleRules (53.22s)
--- PASS: TestAccS3BucketWebsiteConfiguration_RoutingRules_ConditionAndRedirect (81.73s)
```

### CODE REVIEW

File: internal/service/s3/bucket_website_configuration.go [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+						"protocol": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice(s3.Protocol_Values(), false),
+						},
+					},
+				},
+			},
+			"routing_rule": {
+				Type:     schema.TypeSet,
```
Comment (gdavison) [inline internal/service/s3/bucket_website_configuration.go:92]: Is there any ordering to the redirect rules? E.g. if I create rules like

```
`docs/`        => destinationA
`docs/images/` => destinationB
```

does the order in the generated JSON matter?

**Review by gdavison — COMMENTED** [review]
Looks good! One question about the ordering of rules

File: internal/service/s3/bucket_website_configuration.go (hunk 1 above)
Comment (anGie44) [reply internal/service/s3/bucket_website_configuration.go:92]: Ooo that's a good question, and i want to say it _does_ matter so this should really be a List..also the `aws_s3_bucket` resource's `routing_rules` argument doesn't have any 
```				
DiffSuppressFunc: verify.SuppressEquivalentJSONDiffs,
```
which makes me think the List is more appropriate

Comment (anGie44) [reply internal/service/s3/bucket_website_configuration.go]: updated to TypeList in [a90dcdc](https://github.com/hashicorp/terraform-provider-aws/pull/22648/commits/a90dcdcde0b0754d69550612a44e32d7791ec7bc) 👍

**Comment by anGie44** [comment]
Output of acceptance tests after [a90dcdc](https://github.com/hashicorp/terraform-provider-aws/pull/22648/commits/a90dcdcde0b0754d69550612a44e32d7791ec7bc):

```
--- PASS: TestAccS3BucketWebsiteConfiguration_disappears (31.34s)
--- PASS: TestAccS3BucketWebsiteConfiguration_basic (34.80s)
--- PASS: TestAccS3BucketWebsiteConfiguration_RoutingRules_RedirectOnly (34.82s)
--- PASS: TestAccS3BucketWebsiteConfiguration_Redirect (34.86s)
--- PASS: TestAccS3BucketWebsiteConfiguration_update (56.15s)
--- PASS: TestAccS3BucketWebsiteConfiguration_RoutingRules_MultipleRules (57.64s)
--- PASS: TestAccS3BucketWebsiteConfiguration_RoutingRules_ConditionAndRedirect (93.74s)
```

**Review by gdavison — APPROVED** [review]
Looks good! 🚀 

--- PASS: TestAccS3BucketWebsiteConfiguration_disappears (46.52s)
--- PASS: TestAccS3BucketWebsiteConfiguration_basic (52.00s)
--- PASS: TestAccS3BucketWebsiteConfiguration_RoutingRules_RedirectOnly (52.13s)
--- PASS: TestAccS3BucketWebsiteConfiguration_Redirect (52.22s)
--- PASS: TestAccS3BucketWebsiteConfiguration_update (71.78s)
--- PASS: TestAccS3BucketWebsiteConfiguration_RoutingRules_MultipleRules (72.15s)
--- PASS: TestAccS3BucketWebsiteConfiguration_RoutingRules_ConditionAndRedirect (97.63s)

**Pull request merged**